include/iotc.h

```diff
@@ -30,224 +30,192 @@ extern "C" {
- *     can continue to operate.
- * @retval IOTC_EVENT_PROCESS_STOPPED The event processor stopped because the
- *     client application invoked <code>iotc_events_stop()</code> or an
- *     unrecoverable error occurred.
+ * @details Invokes the event processing loop on RTOS or non-OS devices that
+ * must yield for standard tick operations.
+ *
+ * The event engine won't {@link iotc_events_process_tick() process events} when
+ * the Device SDK is in the
+ * {@link ::iotc_state_t IOTC_EVENT_PROCESS_STOPPED state}. If
+ * the function returns IOTC_EVENT_PROCESS_STOPPED,
+ * {@link iotc_shutdown() shutdown} and {@link iotc_initialize() reinitialize}
+ * the Device SDK to process events again.
  */
 extern iotc_state_t iotc_events_process_tick();
 
 /**
  * @brief Shuts down the event engine.
- *
- * @see iotc_events_process_blocking
- * @see iotc_events_process_tick
  */
 extern void iotc_events_stop();
 
 /**
- * @brief Connect to Cloud IoT Core with the provided context.
- *
- * This parameters include a pointer to connection state monitor callback
- * function.
- *
- * <code>iotc_user_callback_t()</code> defines the connection state monitor
- * callback. <code>iotc_user_callback_t()</code> has the following signature:
- *    void foo( iotc_context_handle_t in_context_handle
- *                  , void* data
- *                  , iotc_state_t state )
- *
- * The following are the <code>iotc_user_callback_t()</code> callback function
- * parameters:
- *   - in_context_handle: the context handle provided to
- * <code>iotc_connect()</code>.
- *   - data: a multifunctional structure in the client application. See the <a
- * href="../../../porting_guide.md">porting guide</a> or <code>examples/</code>
- * for more information.
- *   - state: IOTC_STATE_OK if the connection succeeds. See <code><a
- * href="~/include/iotc_error.h">iotc_error.h</a></code> for more error codes.
- *
- * @param [in] iotc_h A context handle created by invoking
- * <code>iotc_create_context()</code>.
+ * @brief Connects to Cloud IoT Core.
+ *
+ * @param [in] iotc_h A {@link iotc_create_context() context handle}.
  * @param [in] username The MQTT username. Cloud IoT Core ignores this
- * parameter.
- * @param [in] password The MQTT password. Cloud IoT Core requires a JWT. Create
- * a JWT with the <code>iotc_create_iotcore_jwt()</code> function.
- * @param [in] client_id The MQTT client identifier. Cloud IoT Core requires a
- *     <a
- * href="https://cloud.google.com/iot/docs/how-tos/mqtt-bridge#device_authentication">device
- * path</a>.
- * @param [in] connection_timeout The number of seconds to keep the socket
- * before CONNACK. If 0, the TCP timeout is used.
- * @param [in] keepalive_timeout The number of seconds that the MQTT service
- * keeps the socket is be kept open without hearing from the client.
- *
- * @see iotc_create_context
- * @see iotc_create_iotcore_jwt
- *
- * @retval IOTC_STATE_OK The connection request is correctly enqueued.
+ *     parameter.
```

Comment:
@atigyi to be fair, this could be re-worded to "Because of a known bug **on some platforms**...". That way it's clear that it's not the fault of your library specifically.

Additionally, this is in line with Google's recommendation for using the MQTT bridge to connect with cloud IoT Core (see https://cloud.google.com/iot/docs/how-tos/mqtt-bridge#configuring_mqtt_clients, item 4)

Reply:
Done